SignInCheckIn/MinistryPlatform.Translation/Repositories/ChildCheckinRepository.cs

```diff
@@ -35,6 +41,40 @@ public List<MpParticipantDto> GetChildrenByEventAndRoom(int eventId, int roomId)
                         SearchTable<MpParticipantDto>("Event_Participants", $"Event_ID_Table.[Event_ID] = {eventId} AND Room_ID_Table.[Room_ID] = {roomId} AND Event_Participants.End_Date IS NULL", columnList);
         }
 
+        public MpEventParticipantDto GetEventParticipantByCallNumber(int eventId, int callNumber)
+        {
+            var apiUserToken = _apiUserRepository.GetToken();
+
+            var columnList = new List<string>
+            {
+                "Event_ID_Table.Event_ID",
+                "Room_ID_Table.Room_ID",
+                "Room_ID_Table.Room_Name",
+                "Event_Participants.Event_Participant_ID",
+                "Event_Participants.Call_Number",
+                "Event_Participants.Time_In",
+                "Event_Participants.Time_Confirmed",
+                "Event_Participants.Checkin_Phone",
+                "Event_Participants.Checkin_Household_ID",
+                "Participant_ID_Table.Participant_ID",
+                "Participant_ID_Table_Contact_ID_Table.Contact_ID",
+                "Participant_ID_Table_Contact_ID_Table.First_Name",
+                "Participant_ID_Table_Contact_ID_Table.Last_Name",
+                "Participant_ID_Table_Contact_ID_Table.Date_of_Birth",
+                "Participant_ID_Table_Contact_ID_Table_Household_ID_Table.Household_ID",
+                "Participation_Status_ID_Table.Participation_Status_ID",
+                "Group_ID_Table.Group_Name"
+            };
+
+            List<MpEventParticipantDto> participants = _ministryPlatformRestRepository.UsingAuthenticationToken(apiUserToken).
+                        SearchTable<MpEventParticipantDto>("Event_Participants", $"Event_ID_Table.[Event_ID] = {eventId} AND Event_Participants.[Call_Number] = {callNumber}", columnList);
+            if (participants.Any())
+            {
+                return participants.First();
+            }
+            return null;
```

Comment:
https://www.lucidchart.com/techblog/2015/08/31/the-worst-mistake-of-computer-science/ don't return null return an exception or something.